gdal/ogr/ogrsf_frmts/flatgeobuf/ogrflatgeobuflayer.cpp

```diff
@@ -767,20 +767,25 @@ OGRLinearRing *OGRFlatGeobufLayer::readLinearRing(const Feature *feature, uint32
 
 OGRErr OGRFlatGeobufLayer::readSimpleCurve(const Feature *feature, uint32_t len, uint32_t offset, OGRSimpleCurve *sc)
 {
-    if (len > std::numeric_limits<int>().max())
+    if (len > feature_max_buffer_size)
```

Comment:
I believe those added checks could be a bit simplied like the following while still avoiding overflow
```
if( offset > feature_max_buffer_size ||
    len > feature_max_buffer_size - offset )
{
    return CPLErrorInvalidSize();
}
const uint32_t offsetLen = len + offset;
```